zcash_client_backend/src/data_api.rs

```diff
@@ -867,6 +867,13 @@ pub trait WalletCommitmentTrees {
         Error = Self::Error,
     >;
 
+    /// Returns the depth of the checkpoint in the tree that can be used to create
+    /// a witness at the anchor having the given number of confirmations.
```

Comment:
An artifact of this method is that some notes that might previously have been spendable may become not so, if the notes were mined between the chosen checkpoint depth and the desired anchor height, and the checkpoint heights are for whatever reason non-contiguous in terms of tree updates (rather than just non-contiguous in checkpoint IDs). Perhaps the checkpoint ID corresponding to this depth should be returned as well?

That does complicate the API a bit though, so maybe instead we just document that this method relies on the caller ensuring that checkpoints exist for every change to the tree, even if there aren't checkpoints for every block.

Reply:
If the note was mined in between, then there'll be a checkpoint for it that is less than the anchor height, and there's not a problem, right?

Reply:
I mean mined between the anchor height and checkpoint ID:
```
chain_tip
 |
 | (10 blocks)
 |
anchor_height
 |
note_mined
 |
checkpoint_id
```
Put another way, the note is selected assuming an anchor will be built to `anchor_height`, while this code now causes it to be built at `checkpoint_id`. If those two are not the same height, there's a potential for errors. We can get around this problem by essentially documenting that we assume no notes ever get mined between `checkpoint_id` and `anchor_height`.

Reply:
If `checkpoint_id` is at a height less than `note_mined`, then the note isn't spendable because the wallet has never observed it as having been mined.

Reply:
In a pairing, we clarified that this is assuming that when a note is witnessed, the block it was witnessed in is being checkpointed. Under that assumption, I agree.

Reply:
Yes that is an assumption I knew we were making. It should be explicitly documented if it isn't already, though.